### PR TITLE
GH-1055 Added support for Pausable

### PR DESCRIPTION
**Pull request by olegz**
Added support for Pausable lifecycle to Binding and its implementation
Added additional actuator endpoints

Resolves spring-cloud/spring-cloud-stream#1055

### CODE REVIEW

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/Binding.java [hunk 1]
```diff
@@ -32,7 +33,7 @@
  * 
  * @see org.springframework.cloud.stream.annotation.EnableBinding
  */
-public interface Binding<T> extends Lifecycle {
+public interface Binding<T> extends Lifecycle, Pausable {
```
Comment (garyrussell) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/Binding.java]: Pausable extends Lifecycle

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java [hunk 2]
```diff
@@ -95,6 +96,20 @@ public final synchronized void stop() {
 			this.lifecycle.stop();
 		}
 	}
+	
+	@Override
+	public final synchronized void pause() {
+		if (this.lifecycle instanceof Pausable) {
+			((Pausable)this.lifecycle).pause();
```
Comment (garyrussell) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java]: space after `)` (checkstyle will complain).

Comment (garyrussell) [inline spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java]: I wonder if the actuator should return an error if it's not a pausable binding?

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/Binding.java (hunk 1 above)
Comment (olegz) [reply spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/Binding.java]: nice!

File: spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java (hunk 2 above)
Comment (olegz) [reply spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java]: it didn't but i'll fix

Comment (olegz) [reply spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinding.java]: may be a WARN for now. If we return something now we'll set the expectations and I am not ready for it yet

**Comment by sobychacko** [comment]
Tested this PR against Kafka binder. LGTM. Merging.

**Pull request closed without merging**